server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadata.java

```diff
@@ -2131,6 +2191,29 @@ public Builder timestampRange(IndexLongFieldRange timestampRange) {
             return this;
         }
 
+        // only for use within this class file where minClusterTransportVersion is not known (e.g., IndexMetadataDiff.apply)
+        Builder eventIngestedRange(IndexLongFieldRange eventIngestedRange) {
+            assert eventIngestedRange != null : "eventIngestedRange cannot be null";
+            this.eventIngestedRange = eventIngestedRange;
+            return this;
+        }
+
+        public Builder eventIngestedRange(IndexLongFieldRange eventIngestedRange, TransportVersion minClusterTransportVersion) {
+            assert eventIngestedRange != null : "eventIngestedRange cannot be null";
+            assert minClusterTransportVersion != null || eventIngestedRange == IndexLongFieldRange.UNKNOWN
+                : "eventIngestedRange should only be UNKNOWN if minClusterTransportVersion is null, but minClusterTransportVersion: "
```

Comment:
I think this should be:

```suggestion
                : "eventIngestedRange must be UNKNOWN when minClusterTransportVersion is null, but minClusterTransportVersion: "
```